libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/MapboxNavigationAccounts.kt

```diff
@@ -36,27 +36,32 @@ class MapboxNavigationAccounts private constructor() : SkuTokenProvider {
             skuGenerator = when (Billing.getInstance(context).getBillingType()) {
                 Billing.BillingModel.MAU -> MauSku(
                     preferences,
-                    TIMER_EXPIRE_AFTER * MAU_TIMER_EXPIRE_THRESHOLD
-                )
-                Billing.BillingModel.TRIPS -> TripsSku(
-                    preferences, TIMER_EXPIRE_AFTER * TRIPS_TIMER_EXPIRE_THRESHOLD,
-                    TRIPS_REQUEST_COUNT_THRESHOLD
+                    TIMER_EXPIRE_AFTER * MAU_TIMER_EXPIRE_THRESHOLD,
+                    context.applicationContext.packageName
                 )
+                Billing.BillingModel.TRIPS -> null
             }
         }
     }
 
-    override fun obtainSkuToken(resourceUrl: String?, querySize: Int): String {
-        val skuToken = skuGenerator?.generateToken() ?: ""
+    @Synchronized
+    override fun obtainUrlWithSkuToken(resourceUrl: String, querySize: Int): String {
+        return skuGenerator?.let { generator ->
+            val skuToken = generator.generateToken() ?: ""
```

Comment:
NIT

```suggestion
            val skuToken = generator.generateToken()
```

as `generateToken` returns empty (`""`) if it fails.

Reply:
Good catch. This is what happens when you copy/paste